gdb/plan.sh

```diff
@@ -1,32 +1,58 @@
 pkg_name=gdb
 pkg_origin=core
-pkg_version=7.11
+pkg_version=7.11.1
 pkg_maintainer="The Habitat Maintainers <humans@habitat.sh>"
-pkg_license=('gplv3')
-pkg_source=http://ftp.gnu.org/gnu/$pkg_name/${pkg_name}-${pkg_version}.tar.xz
-pkg_shasum=7a434116cb630d77bb40776e8f5d3937bed11dea56bafebb4d2bc5dd389fe5c1
-pkg_deps=(core/glibc core/readline core/zlib core/python)
-pkg_build_deps=(core/coreutils core/diffutils core/patch core/make core/gcc core/texinfo)
+pkg_license=('GPL-3.0')
+pkg_description="GDB, the GNU Project debugger, allows you to see what is going on 'inside' another program while it executes -- or what another program was doing at the moment it crashed."
+pkg_upstream_url=https://www.gnu.org/software/gdb/
+pkg_source=http://ftp.gnu.org/gnu/${pkg_name}/${pkg_name}-${pkg_version}.tar.xz
+pkg_shasum=e9216da4e3755e9f414c1aa0026b626251dfc57ffe572a266e98da4f6988fc70
+pkg_deps=(core/glibc core/readline core/zlib core/ncurses core/expat core/python)
+pkg_build_deps=(core/coreutils core/diffutils core/expect core/dejagnu core/patch core/make core/gcc core/texinfo)
 pkg_bin_dirs=(bin)
 pkg_lib_dirs=(lib)
 
 do_build() {
+  CFLAGS="${CFLAGS} -O2 -fstack-protector-strong -Wformat -Werror=format-security "
+  CXXFLAGS="${CXXFLAGS} -O2 -fstack-protector-strong -Wformat -Werror=format-security "
+  CPPFLAGS="${CPPFLAGS} -Wdate-time"
+  LDFLAGS="${LDFLAGS} -Wl,-Bsymbolic-functions -Wl,-z,relro"
+
   ./configure \
-    --prefix=$pkg_prefix \
+    --build=x86_64-linux-gnu \
+    --host=x86_64-linux-gnu \
+    --prefix="${pkg_prefix}" \
+    --sysconfdir=/etc \
+    --localstatedir=/var \
```

Comment:
We need to set this to a package-specific location. Part of the philosophy is that Habitat applications and their dependencies are self-contained and isolated away from the underlying operating system or distribution details.

The general standard for Habitat is that these go in the service directory - `/hab/svc/PKGNAME/{var,etc}`. As an example, [core/openssh](https://github.com/habitat-sh/core-plans/blob/master/openssh/plan.sh#L17-L18) does exactly this with its `configure` in `do_build`.

I know it's different than what we're all used to from Linux distributions, but it's very much on purpose that we do this, and we should expect to configure `core` packages this way.